### PR TITLE
feat: able to generate embeddings and summarization for documents  in a conversation

### LINKED ISSUE #553: AI Agent - Knowledge Assistant - Create an AI agent that can generate abstract for PDF/Word/web links, and accumulate knowledge
Allow user to upload pdf/word or paste web links and then generate a summary, the prompt might be:
* Prompt 示例

1）英文
Provide a summary based on the given content. Output the summary directly without any introductory text. Here is the content: 

{{.question}}

2）中文
需要你根据一段内容写一段摘要，直接输出摘要，不需要提示性文字，内容如下：

{{.question}}

* Required features
1. upload to 微信小程序，and use whole content as {{.question}} if the content is smaller that 10Mi
2. if file is bigger than 10Mi, split it to chunk of paragraphs, then get abstract of each chunk
3. file management, give each user 1G free s3 storage to save user files, and provide upload/delete/download features
5. user can create knowledgebase based on selected files or based on file labels
need content embedding and Q/A embedding?
6. allow to chat/generate article with knowledgebase
7. support to import chat history from wechat, dingtalk, feishu, etc, QQ ...
8. provide APIs to integrate with other tool/product

### PR DESCRIPTION
**Pull request by bjwswang**
#### Which issue(s) this PR fixes

Fixes #553 

#### Special notes for your reviewer

Configurations for the following test :

- filesize: 300kb
- lllm: qwen7b
- embedding: bge
- gpu: 4090

When using swagger document:
image

image

### CODE REVIEW

File: pkg/vectorstore/vectorstore.go [hunk 1]
```diff
@@ -114,7 +114,7 @@ func AddDocuments(ctx context.Context, log logr.Logger, vs *arcadiav1alpha1.Vect
 	log.Info("handle file: add documents to embedder")
 	if store, ok := s.(*PGVectorStore); ok {
 		// now only pgvector support Row-level updates
-		log.Info("handle file: use pgvector, filter out exist documents")
+		log.V(5).Info("handle file: use pgvector, filter out exist documents")
```
Comment (Abirdcfly) [inline pkg/vectorstore/vectorstore.go:117]: v3 is better

File: pkg/vectorstore/vectorstore.go [hunk 2]
```diff
@@ -128,6 +128,6 @@ func AddDocuments(ctx context.Context, log logr.Logger, vs *arcadiav1alpha1.Vect
 	if finish != nil {
 		finish()
 	}
-	log.Info("handle file succeeded")
+	log.V(5).Info("handle file succeeded")
```
Comment (Abirdcfly) [inline pkg/vectorstore/vectorstore.go:131]: v3 is better

File: pkg/appruntime/chain/mpchain.go [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package chain
+
+// TODO: implement this
```
Comment (Abirdcfly) [inline pkg/appruntime/chain/mpchain.go:19]: remove this file？

File: apiserver/service/chat.go [hunk 4]
```diff
@@ -373,4 +418,7 @@ func registerChat(g *gin.RouterGroup, conf config.ServerConfig) {
 	g.POST("/messages/:messageID/references", auth.AuthInterceptor(conf.EnableOIDC, oidc.Verifier, "get", "applications"), requestid.RequestIDInterceptor(), chatService.ReferenceHandler()) // messages reference
 
 	g.POST("/prompt-starter", auth.AuthInterceptor(conf.EnableOIDC, oidc.Verifier, "get", "applications"), requestid.RequestIDInterceptor(), chatService.PromptStartersHandler())
+
+	// upload docs for conversation
+	g.POST("/conversations/docs", auth.AuthInterceptor(conf.EnableOIDC, oidc.Verifier, "get", "applications"), requestid.RequestIDInterceptor(), chatService.ConversationDocs()) // list conversations
```
Comment (Abirdcfly) [inline apiserver/service/chat.go:423]: Remove the comment？

File: pkg/appruntime/chain/mpchain.go (hunk 3 above)
Comment (bjwswang) [reply pkg/appruntime/chain/mpchain.go:19]: we may need this in our application orchestration,so let's keep this.

**Pull request merged**